extensions/sql-database-projects/src/test/projectController.test.ts

```diff
@@ -330,202 +340,215 @@ describe('ProjectsController: import operations', function (): void {
-
-		const projController = new ProjectsController(testContext.apiWrapper.object, new SqlDatabaseProjectTreeViewProvider());
-		await testUtils.shouldThrowSpecificError(async () => await projController.importNewDatabaseProject({ connectionProfile: mockConnectionProfile }), constants.projectLocationRequired);
+		sinon.stub(vscode.window, 'showInputBox').resolves('MyProjectName');
+		sinon.stub(vscode.window, 'showOpenDialog').resolves(undefined);
+		sinon.stub(vscode.window, 'showQuickPick').resolves({ label: constants.file });
+		const spy = sinon.spy(vscode.window, 'showErrorMessage');
+
+		const projController = new ProjectsController(new SqlDatabaseProjectTreeViewProvider());
+		await projController.importNewDatabaseProject({ connectionProfile: mockConnectionProfile });
+		should(spy.calledOnce).be.true('showErrorMessage should have been called');
+		should(spy.calledWith(constants.projectLocationRequired)).be.true(`showErrorMessage not called with expected message '${constants.projectLocationRequired}' Actual '${spy.getCall(0).args[0]}'`);
 	});
 
 	it('Should show error when no location provided with ExtractTarget = SchemaObjectType', async function (): Promise<void> {
-		testContext.apiWrapper.setup(x => x.showInputBox(TypeMoq.It.isAny(), TypeMoq.It.isAny())).returns(() => Promise.resolve('MyProjectName'));
-		testContext.apiWrapper.setup(x => x.showQuickPick(TypeMoq.It.isAny(), TypeMoq.It.isAny(), TypeMoq.It.isAny())).returns(() => Promise.resolve({ label: constants.schemaObjectType }));
-		testContext.apiWrapper.setup(x => x.showOpenDialog(TypeMoq.It.isAny())).returns(() => Promise.resolve(undefined));
-		testContext.apiWrapper.setup(x => x.workspaceFolders()).returns(() => undefined);
-		testContext.apiWrapper.setup(x => x.showErrorMessage(TypeMoq.It.isAny())).returns((s) => { throw new Error(s); });
-
-		const projController = new ProjectsController(testContext.apiWrapper.object, new SqlDatabaseProjectTreeViewProvider());
-		await testUtils.shouldThrowSpecificError(async () => await projController.importNewDatabaseProject({ connectionProfile: mockConnectionProfile }), constants.projectLocationRequired);
+		sinon.stub(vscode.window, 'showInputBox').resolves('MyProjectName');
+		sinon.stub(vscode.window, 'showQuickPick').resolves({ label: constants.schemaObjectType });
+		sinon.stub(vscode.window, 'showOpenDialog').resolves(undefined);
+		const spy = sinon.spy(vscode.window, 'showErrorMessage');
+
+		const projController = new ProjectsController(new SqlDatabaseProjectTreeViewProvider());
+		await projController.importNewDatabaseProject({ connectionProfile: mockConnectionProfile });
+		should(spy.calledOnce).be.true('showErrorMessage should have been called');
+		should(spy.calledWith(constants.projectLocationRequired)).be.true(`showErrorMessage not called with expected message '${constants.projectLocationRequired}' Actual '${spy.getCall(0).args[0]}'`);
```

Comment:
sinon.restore? Or just do for each test function?

Reply:
Responding just to have explicit responses - sinon.restore is called after every test is ran through the afterEach